e_os.h

```diff
@@ -544,6 +544,8 @@ struct servent *getservbyname(const char *name, const char *proto);
 # define CRYPTO_memcmp memcmp
 #endif
 
+#define IF_FAIL_ASSERT(x)     assert(x); if (!(x))
```

Comment:
Yes, I like that, but I'm not convinced you can treat assert that way. It's also not quite right anyway, because it always returns "true" in a prod build!! How about this:

````
#ifdef NDEBUG
# define ossl_assert(x) (int)(x)
#else
__owur static ossl_inline int ossl_assert_int(int expr, const char *exprstr,
                                              const char *file, int line)
{
    if (!expr)
        OPENSSL_die(exprstr, file, line);

    return expr;
}

# define ossl_assert(x) ossl_assert_int((int)(x), "Assertion failed: "#x, \
                                         __FILE__, __LINE__)

#endif
````